src/components/multi-range/multi-range.component.ts

```diff
@@ -0,0 +1,380 @@
+import { html, nothing } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { property, query, queryAll } from 'lit/decorators.js';
+import formControlStyles from '../../styles/form-control.styles.js';
+import ShoelaceElement from '../../internal/shoelace-element.js';
+import styles from './multi-range.styles.js';
+import type { CSSResultGroup, PropertyValues } from 'lit';
+
+const numericSort = function (a: number, b: number): number {
+  return a - b;
+};
+
+const arraysDiffer = function (a: readonly number[], b: readonly number[]): boolean {
+  a ||= [];
+  b ||= [];
+  if (a.length !== b.length) return true;
+  for (let i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) return true;
+  }
+  return false;
+};
+
+/**
+ * @summary Multi-Ranges allow the user to select multiple values within a given range using a slider with multiple handles.
+ * @documentation https://shoelace.style/components/multi-range
+ * @status experimental
+ * @since next
+ *
+ * @slot label - The range's label. Alternatively, you can use the `label` attribute.
+ * @slot help-text - Text that describes how to use the input. Alternatively, you can use the `help-text` attribute.
+ *
+ * @event sl-blur - Emitted when the control loses focus.
+ * @event sl-change - Emitted when an alteration to the control's value is committed by the user.
+ * @event sl-focus - Emitted when the control gains focus.
+ * @event sl-input - Emitted when the control receives input.
+ *
+ * @cssproperty --thumb-size - The size of the thumb.
+ * @cssproperty --tooltip-offset - The vertical distance the tooltip is offset from the track.
+ * @cssproperty --track-color-active - The color of the portion of the track that represents the current value.
+ * @cssproperty --track-color-inactive - The of the portion of the track that represents the remaining value.
+ * @cssproperty --track-height - The height of the track.
+ */
+export default class SlMultiRange extends ShoelaceElement {
+  static styles: CSSResultGroup = [formControlStyles, styles];
+
+  /** The range's label. If you need to display HTML, use the `label` slot instead. */
+  @property() label = '';
+
+  /** The range's help text. If you need to display HTML, use the help-text slot instead. */
+  @property({ attribute: 'help-text' }) helpText = '';
+
+  /** Disables the range. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** The minimum acceptable value of the range. */
+  @property({ type: Number }) min = 0;
+
+  /** The maximum acceptable value of the range. */
+  @property({ type: Number }) max = 100;
+
+  /** The interval at which the range will increase and decrease. */
+  @property({ type: Number }) step = 1;
+
+  /** The preferred placement of the range's tooltip. */
+  @property() tooltip: 'top' | 'bottom' | 'none' = 'top';
+
+  /** The current values of the range */
+  @property({ type: Array })
+  set value(value: readonly number[]) {
+    this.#value = value || [];
```

Comment:
This property uses getters/settings with a backed value to allow for updating the value without triggering a re-render.